### PR TITLE
Update deposit and deposit-all with new fee estimation

### PR DESCRIPTION
**Pull request by ygao76**
## Summary
Use new fee estimation for 
- deposit
- deposit-all
- fees
Use medium priority level as the fee estimation.
## Testing Plan
local
```
yarn start deposit
yarn run v1.22.19
$ yarn build && yarn start:js deposit
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node --expose-gc --inspect=:0 --inspect-publish-uid=http --enable-source-maps bin/run deposit
Insufficient balance: 0.00000000 IRON.  Fee (0.00000365 IRON) + minimum deposit (0.1 IRON) = total required (0.10000365 IRON)
```
```
yarn start deposit -f 0.1
yarn run v1.22.19
$ yarn build && yarn start:js deposit -f 0.1
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node --expose-gc --inspect=:0 --inspect-publish-uid=http --enable-source-maps bin/run deposit -f 0.1
Error reading the fee value, please enter a valid number.
```

```
yarn start fees
yarn run v1.22.19
$ yarn build && yarn start:js fees
$ tsc -b
$ cross-env OCLIF_TS_NODE=0 IRONFISH_DEBUG=1 node --expose-gc --inspect=:0 --inspect-publish-uid=http --enable-source-maps bin/run fees
Fee distribution for last 10 block
percentile 10: 654 ORE
percentile 20: 654 ORE
percentile 30: 654 ORE
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-cli/src/commands/fees.ts [hunk 1]
```diff
@@ -13,32 +14,28 @@ export class FeeCommand extends IronfishCommand {
...
       this.log(
-        `Fee distribution for last ${numOfBlocks} block${numOfBlocks > 1 ? 's' : ''}: (${
-          response.content.startBlock
-        } - ${response.content.endBlock})\np25: ${response.content.p25} ORE\np50: ${
-          response.content.p50
-        } ORE\np75: ${response.content.p75} ORE`,
+        `Fee distribution for last ${JSON.stringify(numOfBlocks)} block\n` +
+          `percentile ${JSON.stringify(low)}: ${response.content.low || ''} ORE\n` +
+          `percentile ${JSON.stringify(medium)}: ${response.content.medium || ''} ORE\n` +
+          `percentile ${JSON.stringify(high)}: ${response.content.high || ''} ORE\n`,
```
Comment (hughy) [inline ironfish-cli/src/commands/fees.ts:38]: nit - the units for the fee rates will be $ORE/kB

File: ironfish-cli/src/commands/deposit-all.ts [hunk 2]
```diff
@@ -208,6 +196,25 @@ export default class DepositAll extends IronfishCommand {
...
+                amount: CurrencyUtils.encode(confirmedBalance),
+                memo: graffiti,
+              },
+            ],
+          })
+          fee = CurrencyUtils.decode(response.content.medium)
+        } catch {
+          fee = 1n
+        }
+      }
```
Comment (hughy) [inline ironfish-cli/src/commands/deposit-all.ts:216]: We could move this out of the while loop -- `fee` will always be non-null after the first iteration.

Or we could update to get a new estimate on each loop

File: ironfish-cli/src/commands/deposit-all.ts [hunk 3]
```diff
@@ -114,11 +94,21 @@ export default class DepositAll extends IronfishCommand {
...
+        fee = parsedFee
+      }
+    }
+
     const { canSend, errorReason } = await verifyCanSend(
       this.client,
       this.api,
       expirationSequenceDelta,
-      fee,
+      fee || 1n,
```
Comment (hughy) [inline ironfish-cli/src/commands/deposit-all.ts:111]: I think we could remove the `fee` argument from `verifyCanSend` -- checking that the fee is non-negative there seems out-of-place

Comment (ygao76) [reply ironfish-cli/src/commands/deposit-all.ts]: removed fee arugment from verifyCanSend.
Add validation for parsed fee. Check if its valid and > 0

File: ironfish-cli/src/commands/deposit-all.ts (hunk 2 above)
Comment (ygao76) [reply ironfish-cli/src/commands/deposit-all.ts]: check `flags.fee` instead. If user did not pass in fee value. We will get a new estimate fee on each loop.

**Pull request merged**